src/toast/scripts/toast_benchmark_satellite

```diff
@@ -0,0 +1,523 @@
+        pix_dist.pointing = ops.pointing
+        pix_dist.shared_flags = ops.binner.shared_flags
+        pix_dist.shared_flag_mask = ops.binner.shared_flag_mask
+        pix_dist.save_pointing = ops.binner.full_pointing
+    pix_dist.apply(data)
+
+    # creates a map and puts it in args.input_map
+    create_input_maps(args.input_map, ops.pointing.nside, rank, log)
+
+    # adds the scan map operator
+    scan_map = toast.ops.ScanHealpix(pixel_dist=pix_dist.pixel_dist, pointing=pix_dist.pointing, file=args.input_map)
+    scan_map.apply(data)
+
+def run_mapmaker(ops, args, tmpls, data):
+    """ 
+    Build up our map-making operation from the pieces- both operators configured from user options and other operators.
+    """
+    ops.binner.pointing = ops.pointing
+    ops.binner.noise_model = ops.default_model.noise_model
+
+    final_bin = None
+    if ops.binner_final.enabled:
+        final_bin = ops.binner_final
+        if ops.pointing_final.enabled:
+            final_bin.pointing = ops.pointing_final
+        else:
+            final_bin.pointing = ops.pointing
+        final_bin.noise_model = ops.default_model.noise_model
+
+    # A simple binned map will be made if an empty list of templates is passed to the mapmaker.
+    tlist = list()
+    if tmpls.baselines.enabled:
+        tlist.append(tmpls.baselines)
+    tmatrix = toast.ops.TemplateMatrix(templates=tlist)
+
+    ops.mapmaker.binning = ops.binner
+    ops.mapmaker.template_matrix = tmatrix
+    ops.mapmaker.map_binning = final_bin
+    ops.mapmaker.det_data = ops.sim_noise.det_data
+
+    # Run the map making
+    ops.mapmaker.apply(data)
+
+    # Write the outputs
+    for prod in ["map", "hits", "cov", "rcond"]:
+        dkey = f"{ops.mapmaker.name}_{prod}"
+        file = os.path.join(args.out_dir, f"{dkey}.fits")
+        toast.pixels_io.write_healpix_fits(data[dkey], file, nest=ops.pointing.nest)
+
+def compute_science_metric(args, global_timer, n_nodes, rank, log):
+    """ 
+    Computes the science metric and stores it.
+    The metric represents the efficiency of the job in a way that is normalized,
+    taking the job size into account
+    """
+    runtime = global_timer.seconds("toast_benchmark_satellite (science work)")
+    prefactor = 1.0e-3
+    kilo_samples = 1.0e-3 * args.total_samples
+    sample_factor = 1.2
+    det_factor = 2.0
+    metric = (
+        prefactor
+        * args.n_detector ** det_factor
+        * kilo_samples ** sample_factor
+        / (n_nodes * runtime)
+    )
+    if rank == 0:
+        msg = f"Science Metric: {prefactor:0.1e} * ({args.n_detector:d}**{det_factor:0.2f}) * ({kilo_samples:0.3e}**{sample_factor:0.3f}) / ({runtime:0.1f} * {n_nodes}) = {metric:0.2f}"
+        log.info("")
+        log.info(msg)
+        log.info("")
+        with open(os.path.join(args.out_dir, "log"), "a") as f:
+            f.write(msg)
+            f.write("\n\n")
+
+def main():
+    env = toast.utils.Environment.get()
```

Comment:
Since this is a benchmark, we should probably always enable function timers.  At this point in the code we could do:
```
env.enable_function_timers()
```